.github/workflows/ci.yaml

```
@@ -36,47 +36,27 @@ jobs:
       COURSIER_CACHE: /tmp/.cache/coursier/v1
     steps:
       - uses: actions/checkout@v2
-      # Work around https://github.com/actions/cache/issues/133#issuecomment-599102035
-      - name: Fix perms on .cargo so we can restore the cache.
-        run: |
-          if [ -e ~/.cargo ] ; then
-            if [ -e "$(which sudo)" ] ; then
-              SUDO=$(which sudo)
-            fi
-            ${SUDO} chown -R $(whoami):$(id -ng) ~/.cargo/
-          fi
       - name: Restore rust cache
-        uses: actions/cache@v1
-        with:
-          key: ${{ github.workflow }}-rust-${{ matrix.target }}-${{ hashFiles('Cargo.lock') }}
-          restore-keys: |
-            ${{ github.workflow }}-rust-${{ matrix.target }}-
-          path: ~/.cargo
-      - name: Restore coursier cache
-        uses: actions/cache@v1
+        uses: actions/cache@v2
         with:
-          key: ${{ github.workflow }}-coursier-${{ hashFiles('java/tests/*.sbt') }}
+          key: rust-${{ matrix.target }}-${{ hashFiles('**/Cargo.*') }}
           restore-keys: |
-            ${{ github.workflow }}-coursier-
-          path: /tmp/.cache/coursier
-      - name: Restore sbt cache
-        uses: actions/cache@v1
+            rust-${{ matrix.target }}-
+          path: |
+            ~/.cargo
+            target
+      - name: Restore Scala caches
+        uses: actions/cache@v2
         with:
-          key: ${{ github.workflow }}-sbt-${{ hashFiles('java/tests/*.sbt') }}
+          key: scala-${{ hashFiles('java/tests/*.sbt') }}
           restore-keys: |
-            ${{ github.workflow }}-sbt-
-          path: ~/.sbt
-      - uses: actions/setup-java@v1
+            coursier-
```

This needs to match the `key` or it will always miss.
```suggestion
            scala-
```

---

Thanks!